Source/RapyutaSimulationPlugins/Private/Robots/RRBaseRobot.cpp

```diff
@@ -584,61 +584,72 @@ void ARRBaseRobot::Tick(float DeltaSeconds)
     }
 }
 
-void ARRBaseRobot::InitPropertiesFromJSON()
+bool ARRBaseRobot::InitPropertiesFromJSON()
 {
+    // Verify [ROSSpawnParameters], which house JSON
+    if (nullptr == ROSSpawnParameters)
+    {
+        // ELSE: This could only be a BP robot & Editor operations might be being done
+        RR_VERIFY_STATIC_BP_ROBOT(this);

```

Comment:
- currently this case is not happen since this func called inside `if (ROSSpawnParameters)`
-  Doesn't this happen if user want to spawn second robot which is dynamically generated from URDF from inside UE?

Reply:
Ah no, all robot instances go through the same flow of calls here, so right though it is already checked by the caller  (as in `ARRBaseRobot::PreInitializeComponents()`), but that's the caller logics only, which this function is not aware of:
* So ideally the check for `ROSSpawnParameters` should be always put inside this function due to its role as providing Json param inputs.
-> This is more robust that way, when `InitPropertiesFromJSON()` will happen to be moved somewhere else.
* Previously, we hadn't made a bool return, but now we have so I figure adding a basic check serves its purpose in `ARRBaseRobot` as the top parent function, which always needs to return `true` before executing the child overriding ones.

Reply:
Ah no, all robot instances go through the same flow of calls here, so right though it is already checked by the caller  (as in `ARRBaseRobot::PreInitializeComponents()`), but that's the caller logics only, which this function is not aware of:
* So ideally the check for `ROSSpawnParameters` should be always put inside this function due to its role as providing Json param inputs.
-> It is more robust that way, when `InitPropertiesFromJSON()` will happen to be moved somewhere else.
* Previously, we hadn't made a bool return, but now we have so I figure adding a basic check serves its purpose in `ARRBaseRobot` as the top parent function, which always needs to return `true` before executing the child overriding ones.

Reply:
Are there any case that robot from URDF with ROSSpawnParameters == nullptr?

Reply:
So, due to essentially instantiated in a ROS-spawned robot only, if sb (like UE devs) spawns it independently in UE (using UE pure spawning) from URDF, yes it will be null.
Also, it is null in case of BP robot as being opened in the Editor

Reply:
I think, then we don't need to put `RR_VERIFY_STATIC_BP_ROBOT` here which print error log since it is not error.

Reply:
Ah right, to be rem